### DISCUSSION TITLE
How to modify object with onsend?

### DISCUSSION THREAD

**Discussion by urdarinda** [fastify/fastify#3844] [Q&A] [answered]
My response body is of object type and I want to modify it in the onSend hook, but seems only a string, a Buffer, a stream, or null is supported

**Comment by metcoder95** [comment, marked as answer]
Hey! The hook you're looking for is `preSerialization` one, see the documentation [here](https://www.fastify.io/docs/latest/Reference/Hooks/#preserialization).

The `preSerialization` hook allows you to modify/replace the payload before is serialized for being served as a response. The `onSend`, on the other hand, receives the payload already serialized and doesn't run any other serialization prior to serving the request back to the client.

See [Lifecycle of the hooks](https://www.fastify.io/docs/latest/Reference/Lifecycle/)
